enhancements/network/on-prem-configurable-lb-dns-stack.md

```diff
@@ -0,0 +1,332 @@
+---
+title: on-prem-configurable-lb-dns-stack
+authors:
+  - "@yboaron"
+reviewers:
+  - "@sttts"
+  - "@deads2k"
+  - "@celebdor"
+  - "@bcrochet"
+  - "@bnemec"
+approvers:
+  - "@"
+creation-date: 2020-02-11
+last-updated: 2020-02-11
+status: implementable
+---
+
+# On-prem: configurable DNS/LB self-hosted stack
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In the current implementation,  the self-hosted DNS/LB stack runs under the MCO umbrella by default in all on-prem platforms and there's no option to configure it.
+Some customers with high production needs usually have their own external Load Balancing and DNS resolution.
+In addition, there are cases in which some traffic used to provide these self-hosted services is disallowed in the customer's network.
+The outcome of such scenarios is that cluster's resources are spent on providing unused services and generating traffic against certain customers network design.
+
+This enhancement suggests a new method for providing the self-hosted stack.
+
+*Note:* as per the OCP plan to move API and Ingress to K8S SLB, this direction should be investigated deeper after MetalLb will be integrated.
+
+## Motivation
+
+Allow customers to adjust the self hosted DNS/LB stack according to the needs they really have instead of providing them in a generic way.
```

Comment:
Do we have the adjustments fully identified? I'm missing clear requirements that are missing today for starters to be able to justify this undertaking.

Reply:
Yep, we have.
I added a table with the required adjustments

Reply:
Which table? The one below? It only lists imeplementations, not requirements. Why should a customer care about self-hosted or not, of node resolution or not? OpenShift 4.x used to be opinionated. This enhancement looks like a departure of that.